src/main/java/be/seeseemelk/mockbukkit/MockChunkData.java

```diff
@@ -0,0 +1,126 @@
+package be.seeseemelk.mockbukkit;
+
+import be.seeseemelk.mockbukkit.block.data.BlockDataMock;
+import org.bukkit.Material;
+import org.bukkit.World;
+import org.bukkit.block.data.BlockData;
+import org.bukkit.generator.ChunkGenerator;
+import org.bukkit.material.MaterialData;
+import org.jetbrains.annotations.NotNull;
+
+public class MockChunkData implements ChunkGenerator.ChunkData
+{
+
+	private final BlockData[][][] blocks;
+
+	private final int maxHeight;
+	private final int minHeight; // unused but will be useful in 1.17/1.18
+
+	public MockChunkData(World world)
+	{
+		this.maxHeight = world.getMaxHeight();
+		this.minHeight = world.getMinHeight();
+		blocks = new BlockData[16][this.maxHeight][16];
+	}
+
+	@Override
+	public int getMaxHeight()
+	{
+		return this.maxHeight;
+	}
+
+	@Override
+	public void setBlock(int x, int y, int z, @NotNull Material material)
+	{
+		this.setBlock(x, y, z, new BlockDataMock(material));
+	}
+
+	@Override
+	public void setBlock(int x, int y, int z, @NotNull MaterialData material)
+	{
+		this.setBlock(x, y, z, new BlockDataMock(material.getItemType()));
+	}
+
+	@Override
+	public void setBlock(int x, int y, int z, @NotNull BlockData blockData)
+	{
+		try
+		{
+			blocks[x][y][z] = blockData;
+		}
+		catch (ArrayIndexOutOfBoundsException ignored)
+		{
+		}
```

Comment:
Don't use a try/catch like that for god sake.
Java is a language which supports the use of `if`-statements to make code run conditionally, you know.